tls/s2n_record_write.c

```diff
@@ -234,6 +234,18 @@ int s2n_record_writev(struct s2n_connection *conn, uint8_t content_type, const s
     uint16_t block_size = 0;
     uint8_t aad_iv[S2N_TLS_MAX_IV_LEN] = { 0 };
 
+    /* Content_type cannot be anything other than the defined values for TLS1.3.
+     * This should never trigger since we control calls to this function.
+     *= https://tools.ietf.org/rfc/rfc8446#5
+     *# Implementations MUST NOT send record types not defined in this
+     *# document unless negotiated by some extension.
+     */
+    if (conn->actual_protocol_version == S2N_TLS13) {
```

Comment:
We try to use ">=" for version checks instead of "==". If there's a theoretical TLS1.4, it will probably build off of TLS1.3 instead of an earlier version :)